cmplxfoil/CMPLXFOIL.py

```diff
@@ -0,0 +1,1120 @@
+
+        Parameters
+        ----------
+        fileName : str
+            File name for saved pkl file (".pkl" will be automatically appended).
+        """
+        fileName += ".pkl"
+        with open(fileName, "wb") as f:
+            pkl.dump(self.sliceData, f, protocol=pkl.HIGHEST_PROTOCOL)
+
+    def checkSolutionFailure(self, aeroProblem, funcs):
+        """Take in a an aeroProblem and check for failure.
+
+        Then append the fail flag in funcs. Information regarding whether or not the last analysis with the aeroProblem
+        was sucessful is included. This information is included as "funcs['fail']". If the 'fail' entry already exits in
+        the dictionary the following operation is performed:
+
+        funcs['fail'] = funcs['fail'] or <did this problem fail>
+
+        In other words, if any one problem fails, the funcs['fail'] entry will be True. This information can then be
+        used directly in multiPointSparse. For direct interface with pyOptSparse the fail flag needs to be returned
+        separately from the funcs.
+
+        Parameters
+        ----------
+        aeroProblem : pyAero_problem class
+            The aerodynamic problem to to get the solution for
+        funcs : dict
+            Dictionary into which the functions are saved.
+        """
+
+        self.setAeroProblem(aeroProblem)
+        # We also add the fail flag into the funcs dictionary. If fail is already there, we just logically 'or' what was
+        # there. Otherwise we add a new entry.
+        failFlag = self.curAP.solveFailed or self.curAP.fatalFail
+        if "fail" in funcs:
+            funcs["fail"] = funcs["fail"] or failFlag
+        else:
+            funcs["fail"] = failFlag
+
+    def checkAdjointFailure(self, aeroProblem, funcsSens):
+        """
+        Pass through to checkSolutionFailure to maintain the same interface as ADFLOW.
+
+        Take in an aeroProblem and check for adjoint failure, Then append the
+        fail flag in funcsSens. Information regarding whether or not the
+        last analysis with the aeroProblem was sucessful is
+        included. This information is included as "funcsSens['fail']". If
+        the 'fail' entry already exits in the dictionary the following
+        operation is performed:
+
+        funcsSens['fail'] = funcsSens['fail'] or <did this problem fail>
+
+        In other words, if any one problem fails, the funcsSens['fail']
+        entry will be True. This information can then be used
+        directly in multiPointSparse. For direct interface with pyOptSparse
+        the fail flag needs to be returned separately from the funcs.
+
+        Parameters
+        ----------
+        aeroProblem : pyAero_problem class
+            The aerodynamic problem to to get the solution for
+        funcsSens : dict
+            Dictionary into which the functions are saved.
+        """
+        self.checkSolutionFailure(aeroProblem, funcsSens)
```

Comment:
Is this function actually checking the sensitivity failure or does it check if the primal fails when computing the sensitivity? Since we use FD / CS a failure in the primal solution will mean a failed derivative, but if that is the case it may be worth an explanatory comment here.